WebRequest deprecation (#20310)
dotnet/docs: includes/core-changes/networking/5.0/webrequest-deprecate.md

> ### WebRequest, WebClient and ServicePoint are deprecated

Reviewer: ```suggestion
### WebRequest, WebClient, and ServicePoint are deprecated
```

> <xref:System.Net.WebRequest>, <xref:System.Net.WebClient> and <xref:System.Net.ServicePoint> classes was marked as deprecate.

Reviewer: ```suggestion
<xref:System.Net.WebRequest>, <xref:System.Net.WebClient>, and <xref:System.Net.ServicePoint> classes was marked as deprecated.
```

> <xref:System.Net.WebRequest>, <xref:System.Net.WebClient>, and <xref:System.Net.ServicePoint> classes was marked as deprecated.

Reviewer 2: ```suggestion
<xref:System.Net.WebRequest>, <xref:System.Net.WebClient>, and <xref:System.Net.ServicePoint> classes are deprecated.
```

#### Version introduced

5.0

#### Change description

> <xref:System.Net.WebRequest>, <xref:System.Net.WebClient> and <xref:System.Net.ServicePoint> classes was marked as deprecate. The classes are still available but they are not recomentded for new development. Only construction methods were marked with ObsoleteAttibute to reduce number of analyser warnings.

Reviewer: ```suggestion
<xref:System.Net.WebRequest>, <xref:System.Net.WebClient>, and <xref:System.Net.ServicePoint> classes was marked as deprecated. The classes are still available but they are not recommended for new development. Only construction methods were marked with ObsoleteAttribute to reduce number of analyzer warnings.
```

> <xref:System.Net.WebRequest>, <xref:System.Net.WebClient>, and <xref:System.Net.ServicePoint> classes was marked as deprecated. The classes are still available but they are not recommended for new development. Only construction methods were marked with ObsoleteAttribute to reduce number of analyzer warnings.

Reviewer 2: ```suggestion
The <xref:System.Net.WebRequest>, <xref:System.Net.WebClient>, and <xref:System.Net.ServicePoint> classes were added to .NET Core in version 2.0 for backward compatibility. However, they introduced several runtime breaking changes, for example, `WebRequest.GetRequestStream` allocates memory for the whole response, and `WebClient.CancelAsync` doesn't always cancel immediately.

Starting in .NET 5.0, the <xref:System.Net.WebRequest>, <xref:System.Net.WebClient>, and <xref:System.Net.ServicePoint> classes are deprecated. The classes are still available, but they are not recommended for new development. To reduce the number of analyzer warnings, only construction methods are decorated with the <xref:System.ObsoleteAttribute> attribute.
```
